packages/cli/src/ResponseHelper.ts

```diff
@@ -15,44 +16,69 @@ import type {
 	IWorkflowDb,
 } from './Interfaces';
 
+const inDevelopment = !process.env.NODE_ENV || process.env.NODE_ENV === 'development';
+
 /**
  * Special Error which allows to return also an error code and http status code
- *
- * @class ResponseError
- * @extends {Error}
  */
-export class ResponseError extends Error {
-	// The HTTP status code of  response
-	httpStatusCode?: number;
-
-	// The error code in the response
-	errorCode?: number;
-
-	// The error hint the response
-	hint?: string;
-
+abstract class ResponseError extends Error {
 	/**
 	 * Creates an instance of ResponseError.
 	 * Must be used inside a block with `ResponseHelper.send()`.
-	 *
-	 * @param {string} message The error message
-	 * @param {number} [errorCode] The error code which can be used by frontend to identify the actual error
-	 * @param {number} [httpStatusCode] The HTTP status code the response should have
-	 * @param {string} [hint] The error hint to provide a context (webhook related)
 	 */
-	constructor(message: string, errorCode?: number, httpStatusCode?: number, hint?: string) {
+	constructor(
+		message: string,
+		// The HTTP status code of  response
+		readonly httpStatusCode: number,
+		// The error code in the response
+		readonly errorCode: number = httpStatusCode,
+		// The error hint the response
+		readonly hint: string | undefined = undefined,
+	) {
 		super(message);
 		this.name = 'ResponseError';
+	}
+}
 
-		if (errorCode) {
-			this.errorCode = errorCode;
-		}
-		if (httpStatusCode) {
-			this.httpStatusCode = httpStatusCode;
-		}
-		if (hint) {
-			this.hint = hint;
-		}
+export class BadRequestError extends ResponseError {
+	constructor(message: string) {
+		super(message, 400);
+	}
+}
+
+export class AuthError extends ResponseError {
+	constructor(message: string) {
+		super(message, 401);
+	}
+}
+
+export class UnauthorizedError extends ResponseError {
+	constructor(message: string, hint: string | undefined = undefined) {
+		super(message, 403, 403, hint);
+	}
+}
+
+export class NotFoundError extends ResponseError {
+	constructor(message: string, hint: string | undefined = undefined) {
+		super(message, 404, 404, hint);
+	}
+}
+
+export class ConflictError extends ResponseError {
+	constructor(message: string, hint: string | undefined = undefined) {
+		super(message, 409, 409, hint);
+	}
+}
+
+export class InternalServerError extends ResponseError {
+	constructor(message: string, errorCode = 500) {
+		super(message, 500, errorCode);
```

Comment:
The HTTP status code is the third argument, not the second, so I guess it should be `super(message, errorCode, 500);` instead, right?

Reply:
it's the second parameter now because it's non-optional now, and has to come before the optional params.